### DISCUSSION TITLE
Building `FastAPI` with `buildroot` `2022.05` (with `flit`support) fails

### DISCUSSION THREAD

**Discussion by maovidal** [fastapi/fastapi#8361] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Here are my two `Buildroot's` files for a package named `python-fastapi`:

[Config.in]
config BR2_PACKAGE_PYTHON_FASTAPI
	bool "python-fastapi"
	help
	  FastAPI framework, high performance, easy to learn, fast to code, ready for production

	  https://github.com/tiangolo/fastapi

[python-fastapi.mk]
################################################################################
#
# python-fastapi
#
################################################################################

PYTHON_FASTAPI_VERSION = 0.79.0
PYTHON_FASTAPI_SOURCE = fastapi-$(PYTHON_FASTAPI_VERSION).tar.gz
PYTHON_FASTAPI_SITE = https://files.pythonhosted.org/packages/ae/ed/ea37410618f1c206ba857d391d7b2de6de7a758ea586662aef77c945d3b4
PYTHON_FASTAPI_LICENSE = MIT
PYTHON_FASTAPI_LICENSE_FILES = LICENSE
PYTHON_FASTAPI_SETUP_TYPE = flit

$(eval $(python-package))
```

### Description

The last `2022.05` version of [buildroot](https://buildroot.org) has support to build python packages that use `flit`.

I don't know much about the details of how a python is packaged (and less about `flit`), but I understand that `FastAPI` provides a `pyproject.toml` file where the package details are contained.

I was able to successfully build the package once I changed this part of that file:

```
[build-system]
 requires = ["flit"]
 build-backend = "flit.buildapi"
```

To this one:

```
[build-system]
 requires = ["flit"]
 build-backend = "flit_core.buildapi"
```

But I'm not sure if that breaks it for someone else.

### Operating System

Linux

### FastAPI Version

0.79.0

### Python Version

3.10.4

**Comment by JarroVGIT** [comment]
I am unfamiliar with buildroot, but I see you can also build from pypi:

```
# 18.9.3. Generating a python-package from a PyPI repository
If the Python package for which you would like to create a Buildroot package is available on PyPI, you may want to use the scanpypi tool located in utils/ to automate the process.

You can find the list of existing PyPI packages [here](https://pypi.python.org/).

scanpypi requires Python’s setuptools package to be installed on your host.

When at the root of your buildroot directory just do :

utils/scanpypi foo bar -o package
This will generate packages python-foo and python-bar in the package folder if they exist on https://pypi.python.org/.

Find the external python modules menu and insert your package inside. Keep in mind that the items inside a menu should be in alphabetical order.
```

Wouldn't that make it a lot easier? FastAPI is packaged and published with every release to PyPi.

**Comment by maovidal** [comment]
The `scanpypi` utility just generates  the files `config.ini`, `*.mk` (and hashes) from the information about the package available at `pypy` website. The result is  similar to the content I included in the example code.

Those files are used by `Buildroot` to build the package. `scanpypi` does not build the package.

**Comment by JarroVGIT** [comment]
Ah okay then I'm sorry, I have no knowledge of buildroot so I wouldn't know. All I know is that FastAPI is a python package and those are generally not 'built' unless you have some C modules (which FastAPI doesn't have).

**Comment by maovidal** [comment]
No worries @JarroVGIT , thank you for taking the time to review this.
Actually the package can be built with the little change on the .toml file I mentioned on the Description. But I also have to learn about `flit` to propose a fix.
Thanks again!

**Comment by iudeen** [comment, marked as answer]
Once FastAPI drops support for python 3.6, it'd most likely move away from `flit` towards [Hatch](https://hatch.pypa.io/latest/). Will it solve your problem?

**Comment by maovidal** [comment]
It seems `buildroot` currently does not have support for `Hatch`. But I noted that `pypa` is on its way to embrace it, so that's something to work on the `buildroot` side.
Thank you for the heads up @iudeen
